src/tap/data/ResultSetTableIterator.java

```diff
@@ -731,7 +731,7 @@ else if (colValue instanceof Boolean)
 			else if (colType != null && colValue != null && colType.type == DBDatatype.CHAR && (colType.length == 1 || colType.length <= 0) && colValue instanceof String)
 				colValue = ((String)colValue).charAt(0);
 			// if the column value is a geometrical object, it must be serialized in STC-S:
-			else if (translator != null && colType != null && colType.isGeometry()){
+			else if ((translator != null && colType != null && colType.isGeometry()) || colValue instanceof byte[]) {
```

Comment:
Ok. So the real problem comes from the bad automatic typing of these processed columns/sub-query columns. Do you have specific examples in mind so that we can work on improving that aspect in the library?